spec/support/test_helpers/teacher_training_public_api_helper.rb

```diff
@@ -27,20 +27,20 @@ def stub_teacher_training_api_provider(provider_code:, recruitment_cycle_year: R
     )
   end
 
-  def stub_teacher_training_api_course_with_site(recruitment_cycle_year: RecruitmentCycle.current_year, provider_code:, course_code:, site_code:, vacancy_status: 'full_time_vacancies', course_attributes: [], site_attributes: [])
-    course_attributes = course_attributes.any? ? [course_attributes.first.merge(code: course_code)] : [{ code: course_code }]
+  def stub_teacher_training_api_course_with_site(recruitment_cycle_year: RecruitmentCycle.current_year, provider_code:, course_code:, site_code:, vacancy_status: 'full_time_vacancies', course_attributes: [], site_attributes: [], course_uuid: SecureRandom.uuid)
+    course_attributes = course_attributes.any? ? [course_attributes.first.merge(code: course_code, uuid: course_uuid)] : [{ code: course_code, uuid: course_uuid }]
     site_attributes = site_attributes.any? ? [site_attributes.first.merge(code: site_code)] : [{ code: site_code }]
     stub_teacher_training_api_courses(recruitment_cycle_year: recruitment_cycle_year, provider_code: provider_code, specified_attributes: course_attributes)
     stub_teacher_training_api_sites(recruitment_cycle_year: recruitment_cycle_year, provider_code: provider_code, course_code: course_code, specified_attributes: site_attributes, vacancy_status: vacancy_status)
   end
 
-  def stub_teacher_training_api_course(recruitment_cycle_year: RecruitmentCycle.current_year, provider_code:, course_code:, specified_attributes: [])
-    response_body = build_response_body('course_single_response.json', specified_attributes.merge(code: course_code))
+  def stub_teacher_training_api_course(recruitment_cycle_year: RecruitmentCycle.current_year, provider_code:, course_code:, specified_attributes: [], course_uuid: SecureRandom.uuid)
+    response_body = build_response_body('course_single_response.json', specified_attributes.merge(code: course_code, uuid: course_uuid))
     stub_teacher_training_single_api_request("#{ENV.fetch('TEACHER_TRAINING_API_BASE_URL')}recruitment_cycles/#{recruitment_cycle_year}/providers/#{provider_code}/courses/#{course_code}", response_body)
   end
 
-  def stub_teacher_training_api_courses(recruitment_cycle_year: RecruitmentCycle.current_year, provider_code:, specified_attributes: [])
-    response_body = build_response_body('course_list_response.json', specified_attributes)
+  def stub_teacher_training_api_courses(recruitment_cycle_year: RecruitmentCycle.current_year, provider_code:, specified_attributes: [], course_uuid: SecureRandom.uuid)
+    response_body = build_response_body('course_list_response.json', specified_attributes.first.merge(uuid: course_uuid) )
```

Comment:
I need advice on how to restructure this method. It will break if we add multiple courses as part of a single stub.